app/src/main/java/com/github/sdp/mediato/ui/BaseProfileFragment.java

```diff
@@ -75,13 +81,49 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     return view;
   }
 
-  abstract CollectionListAdapter setupCollections(RecyclerView recyclerView);
-
+  /**
+   * Method to define how the collections on the profile are set up.
+   * @param recyclerView the collections recycler view to set up
+   * @param collections the list of collections to display
+   * @return the set up collection list adapter
+   */
+  abstract CollectionListAdapter setupCollections(RecyclerView recyclerView, List<Collection> collections);
+
+  /**
+   * Method to make the collection list adapter observe collections in the view model.
+   * @param collectionsAdapter the collection list adapter that should observe the collections
+   */
   public void observeCollections(CollectionListAdapter collectionsAdapter) {
     viewModel.getCollectionsLiveData()
         .observe(getViewLifecycleOwner(), collections -> collectionsAdapter.notifyDataSetChanged());
   }
 
+
+  /**
+   * Method to download collections from the database with retrying.
+   * @param count how many retries were already done
+   * @return the future list of collections that was downloaded
+   */
+  public CompletableFuture<List<Collection>> fetchCollectionsFromDatabaseWithRetry(int count) {
```

Comment:
Is there a reason why this method is defined here instead of the database classes ?

Reply:
Yes, because this function is interacting directly with the view model of the profile page. It uses the username set in the view model and sets the collections in the view model, so it feels more appropriate to have it here instead of mixing that up with pure database functions.